ext/mbstring/libmbfl/mbfl/mbfl_encoding.c

```diff
@@ -359,7 +363,7 @@ const mbfl_encoding *mbfl_name2encoding_ex(const char *name, size_t name_len)
 	for (encoding = mbfl_encoding_ptr_list; *encoding; encoding++) {
 		if ((*encoding)->aliases) {
 			for (const char **alias = (*encoding)->aliases; *alias; alias++) {
-				if (strcasecmp(*alias, name) == 0) {
+				if (strncasecmp(name, *alias, name_len) == 0 && strlen(*alias) == name_len) {
```

Comment:
The following would be cheaper.
We only have to check the NUL terminator because all previous characters matched.

```suggestion
				if (strncasecmp(name, *alias, name_len) == 0 && (*alias)[name_len] == '\0') {
```